lib/membrane_opus/parser/frame_lengths.ex

```diff
@@ -1,91 +1,146 @@
 defmodule Membrane.Opus.Parser.FrameLengths do
   @moduledoc false
-  # Helper for Membrane.Opus.Parser for determining frame lengths
+  # Helper for Membrane.Opus.Parser for working with frame lengths
 
-  # given a frame length, return opus binary encoded length
-  @spec encode_frame_length(non_neg_integer) :: binary
-  def encode_frame_length(length) do
-    if length < 252 do
-      <<length::size(8)>>
-    else
-      <<252 + rem(length - 252, 4)::size(8), div(length - 252, 4)::size(8)>>
-    end
-  end
-
-  # returns ordered list of frame lengths and header length
-  @spec parse_frame_lengths(0..3, binary, boolean) :: {[non_neg_integer], pos_integer}
-  def parse_frame_lengths(frame_packing, data_without_toc, self_delimiting?) do
+  @spec parse(frame_packing :: 0..3, data :: binary, self_delimiting? :: boolean) ::
+          {header_size :: pos_integer, frame_lengths :: [non_neg_integer],
+           padding_size :: non_neg_integer}
+          | :cont
+          | :error
```

Comment:
if we return `:error`, we should return `:ok` tuple in all non-error cases. Also, what does `:cont` mean? It seems it's never matched on

Reply:
`:cont` is "there's an error but it means that we should just get the next packet". what do you think that should look like in terms of return vals here?

Reply:
I meant something like

```suggestion
  @spec parse(frame_packing :: 0..3, data :: binary, self_delimiting? :: boolean) ::
          {:ok, header_size :: pos_integer, frame_lengths :: [non_neg_integer],
           padding_size :: non_neg_integer}
          | {:ok, :cont}
          | :error
```

The `:cont` itself is a cool idea, but it's never used anywhere.